Zend/zend_compile.c

```diff
@@ -2483,7 +2498,16 @@ void zend_emit_final_return(bool return_one) /* {{{ */
 
 	if ((CG(active_op_array)->fn_flags & ZEND_ACC_HAS_RETURN_TYPE)
 			&& !(CG(active_op_array)->fn_flags & ZEND_ACC_GENERATOR)) {
-		zend_emit_return_type_check(NULL, CG(active_op_array)->arg_info - 1, 1);
+		zend_arg_info *return_info = CG(active_op_array)->arg_info - 1;
+
+		zend_type type = return_info->type;
+
+		if (ZEND_TYPE_IS_SET(type) && ZEND_TYPE_CONTAINS_CODE(type, IS_NEVER)) {
```

Comment:
The ZEND_TYPE_IS_SET check is unnecessary. At which point you can save the `type` variable as well.